app/models/target.rb

```diff
@@ -28,11 +28,32 @@ class Target < ApplicationRecord
 
   validate :target_limit, on: :create
 
+  acts_as_mappable :default_units => :meters,
+                   :default_formula => :sphere,
+                   :distance_field_name => :distance,
+                   :lat_column_name => :latitude,
+                   :lng_column_name => :longitude
+
+  after_create :create_matches
+
+  has_many :matches, dependent: :destroy
+
   private
 
   def target_limit
     return unless user.targets.count == USER_MAX_TARGET_LIMIT
 
     errors.add(:targets, I18n.t('api.errors.target_limit'))
   end
+
+  def create_matches
+    target_query = TargetsQuery.new
+    targets = target_query.get_close_targets(self)
+
+    targets.find_each do |target|
+      new_match = Hash(first_user_id: user_id, second_user_id: target.user_id, target_id: target.id)
+      match_form = MatchForm.new(match_attributes: new_match)
```

Comment:
```suggestion
      match_form = MatchForm.new(match_attributes: self)
```

Reply:
Not possible, raised ArgumentError exception.